clinvar/src/main/scala/org/broadinstitute/monster/etl/clinvar/models/intermediate/VariationArchive.scala

```diff
@@ -247,60 +239,92 @@ object VariationArchive {
           // NOTE: Because trait mappings link to the numeric ID for each SCV,
           // but we use the accession as the PK, we need to do a little bit
           // of post-processing on the mappings.
-          val scvId = rawScv
-            .extract("@ID")
-            .getOrElse {
-              throw new IllegalStateException(s"Found an SCV with no numeric ID: $scv")
-            }
-            .str
-          scvIdToAccession.update(scvId, scv.id)
+          val scvId = SCV.extractNumericId(rawScv)
+          scvIdToAccession.update(scvId, scvAccessionId)
           val relevantMappings = mappingsByScvId.getOrElse(scvId, Array.empty)
 
           rawScv.extract("TraitSet").foreach { rawTraitSet =>
-            val traitSet = SCVTraitSet.fromRawAssertionSet(scv, rawTraitSet)
             val traitCounter = new AtomicInteger(0)
+            val currentScvTraitIds = new mutable.ArrayBuffer[String]()
             MsgTransformations.popAsArray(rawTraitSet, "Trait").foreach { rawTrait =>
               val scvTrait = SCVTrait.fromRawTrait(
-                traitSet,
+                // the setId is the same as the scv.id when extracting from assertions
+                scvAccessionId,
                 traitsWithoutContent,
                 relevantMappings,
                 traitCounter,
                 rawTrait
               )
               scvTraits.append(WithContent.attachContent(scvTrait, rawTrait))
+              currentScvTraitIds.append(scvTrait.id)
             }
+            val traitSet = SCVTraitSet.fromRawAssertionSet(
+              scvAccessionId,
+              rawTraitSet,
+              currentScvTraitIds.toArray
+            )
```

Comment:
We shouldn't need to do any "which SCVTraitSet / SCVTraits are relevant for this SCV?" searches, because we already have the answer here. I would update the logic so we can pass `traitSet` and its corresponding traits into the SCV helper functions directly.